scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -436,112 +444,237 @@
+                ;; the pending-jobs atom is repopulated
                 @(d/transact
                    conn
-                   (mapcat (fn [task-info]
-                             [[:job/allowed-to-start? (:job-id task-info)]
-                              {:db/id (d/tempid :db.part/user)
-                               :job/_instance (:job-id task-info)
-                               :instance/task-id (:task-id task-info)
-                               :instance/hostname (:hostname offer)
-                               :instance/start-time (now)
-                               ;; NB command executor uses the task-id
-                               ;; as the executor-id
-                               :instance/executor-id (get-in
-                                                       task-info
-                                                       [:executor :executor-id]
-                                                       (:task-id task-info))
-                               :instance/slave-id (:slave-id offer)
-                               :instance/progress 0
-                               :instance/status :instance.status/unknown
-                               :instance/preempted? false}])
-                           task-infos)))
-              (log/info "Matched offer" offer "to tasks" task-infos)
-              ;; This launch-tasks MUST happen after the above transaction in
-              ;; order to allow a transaction failure (due to failed preconditions)
-              ;; to block the launch
-              (meters/mark! scheduler-offer-matched (count (:ids offer)))
-              (meters/mark! matched-tasks (count task-infos))
-              (histograms/update! number-tasks-matched (count task-infos))
-              (meters/mark! matched-tasks-cpus (:cpus match-resource-requirements))
-              (meters/mark! matched-tasks-mem (:mem match-resource-requirements))
-              (mesos/launch-tasks
-                driver
-                (:ids offer)
-                (mapv #(dissoc % :job-id :num-ports) task-infos))))))
+                   (vec (apply concat upgrade-txns task-txns)))
+                (log/info "Launching" (count task-txns) "tasks")
+                (log/info "Upgrading" (count (:upgrade-backfill processed-matches)) "tasks from backfilled to proper")
+                (log/debug "Matched tasks" task-txns)
+                ;; This launch-tasks MUST happen after the above transaction in
+                ;; order to allow a transaction failure (due to failed preconditions)
+                ;; to block the launch
+                (meters/mark! scheduler-offer-matched
+                              (->> matches
+                                   (mapcat (comp :id :offer :leases))
+                                   (distinct)
+                                   (count)))
+                (meters/mark! matched-tasks (count task-txns))
```

Comment:
So about these:  I've started on this, but realized that in order to record these metrics, we're probably best off redoing the way that  process-matches-for-backfill works.   

For example, "upgraded" is a list of task id's... to get total resources requested (in terms of jobs) from that one we'd have to go back to Datomic to match the task id's to jobs, and then sum the attributes of the jobs.  If we're going to do that it would be more efficient to build up a map of jobs->tasks in the main loop in process-matches-for-backfill.

And for fully-processed and backfilled... these at least are job id's, but again, if we're just going to have to go back to datomic to get more attributes of the jobs in order to do resource totals, it probably makes more sense for the main loop  in process-matches-for-backfill to be accumulating whole jobs rather than just the ids.

Can definitely be done but I think this is worth me checking in with you to ask:  Still worth it?